### PR TITLE
Add psvDebug, Ctrl, Touch samples

### PR DESCRIPTION
**Pull request by yne**
This PR add the following:
- Codebase cleanup
- Add basic ANSI escape support to the psvDebug library (no bold, underline, blink support)
- Add psvDebug sample (with colorization, and placement)
- Add sceCtrl sample
- Add sceTouch sample

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
 # vitasdk code samples
 
-Every directory contains a single sample. To build all samples, add the toolchain `bin/` directory to your `$PATH`, then run `./build.sh`. To clean all samples, run `./clean.sh`. To build a single sample, `cd` to its directory and type `make`.
+## Prerequists
```
Comment (xyzz) [inline README.md]: Prerequisites

File: README.md [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-* `hello_world`: A minimal Makefile-based C project
-* `hello_cpp_world`: A minimal Makefile-based C++ project
+Every directory contains a single sample. `cd` to its directory and use the `make` command to build it.
+
+## Building everything
+
+Use the following command to build every samples
+
+```
+for f in */Makefile; do make -C ${f%/*} all; done
```
Comment (xyzz) [inline README.md]: this is outdated now, just run `build.sh`?

Comment (yne) [inline README.md]: As said on https://github.com/vitasdk/samples/pull/4

> Basically, when this repository will have dozens of samples (I hope it will),  I seriously doubt that people will build all of them. They will simply go the the sample they are interested in an build-install-run it.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+## Prerequists
 
-Always use the `*_debug.bat` (e.g. `run_homebrew_unity_debug.bat`) launcher to get debug output from Vita.
+In order to build a vita sample, you need to add the toolchain `bin/` directory to your `$PATH`.
 
-## List of samples
+## Building
 
-* `hello_world`: A minimal Makefile-based C project
-* `hello_cpp_world`: A minimal Makefile-based C++ project
```
Comment (xyzz) [inline README.md]: Why did you delete this? It would be nice to have short sample descriptions here, with optionally a longer README.md in the sample directory.

File: README.md (hunk 1 above)
Comment (yne) [inline README.md]: Yup, sorry

File: README.md (hunk 3 above)
Comment (yne) [inline README.md]: Because nobody will remember to update the main README when they will add/change a sample (for example, the net_http sample were already missing ;-) ).

:+1: for the per-sample README

We could just explain in this main README that every samples have they own README
So we don't have to update the main one each time.

**Comment by xyzz** [comment]
ok so just fix the typo and I'll merge it

**Comment by yne** [comment]
I've fixed the typo, and added a new RTC sample to match the vita-header addition.
